### DISCUSSION TITLE
Default compile_commands.json lookup locations

### DISCUSSION THREAD

**Discussion by madhurajith** [microsoft/vscode-cpptools#9149] [Q&A]
Are there any default locations vscode-cpptools look for the compile_commands.json file?

For example i could find this documentation for clangd, but i can't find anything like that for vscode-cpptools.

https://clangd.llvm.org/installation.html#project-setup

There the compile_commands.json can be put in any parent folder or the build sub folder of any of parent folders.

Is this the same for vscode-cpptools?

**Comment by sean-mcmanus** [comment]
If compile_commands.json is detected under the workspace folder, you should get a prompt UI to use that, but we won't automatically use that -- we do automatically use a configuration provider in some cases though. We don't search up the parent folders -- you could file a issue request for us to add that.

**Reply by madhurajith** [reply]
Looks like compile_commands.json is atleast looked up in the ${workspace}/build folder. I can see the following message in the output window,

"build/compile_commands.json" could not be found. 'includePath' from c_cpp_properties.json in folder 'x' will be used instead.

**Comment by bobbrow** [comment]
You can set your own default location by using the `C_Cpp.default.compileCommands` setting.
